### PR TITLE
fix/error display name

### PR DESCRIPTION
**Pull request by juliushaertl**
- **fix: Set displayname for error response**
- **chore: Remove compatibility check for display name**

We do not store the display name in the database as it might be used by native WebDAV locking for weird identifiers from the webdav clients. However with that we need to enhance the lock object with the actual display name before using it in an error response.

### CODE REVIEW

File: lib/Service/LockService.php [hunk 1]
```diff
@@ -344,13 +344,7 @@ public function injectMetadata(FileLock $lock): FileLock {
...
 		}
 		if ($lock->getType() === ILock::TYPE_TOKEN) {
-			if (!method_exists($this->userManager, 'getDisplayName')) {
-				// TODO: Remove once 25 is the minimum supported version
-				$user = $this->userManager->get($lock->getOwner());
-				$displayName = $user ? $user->getDisplayName(): $lock->getDisplayName();
-			} else {
-				$displayName = $this->userManager->getDisplayName($lock->getOwner()) ?? $lock->getDisplayName();
-			}
+			$displayName = $this->userManager->getDisplayName($lock->getOwner()) ?? $lock->getDisplayName();
```
Comment (come-nc) [inline lib/Service/LockService.php:347]: The same TODO is on line 337 as well.
See e35d7ea0ad528e20fca57c32e1b401cb7262970a
See also https://github.com/nextcloud/files_lock/blob/main/lib/Tools/Db/ExtendedQueryBuilder.php#L62

Comment (juliushaertl) [reply lib/Service/LockService.php:342]: Thanks, also adjusted

**Review by come-nc — APPROVED** [review]
Confirmed working.

**Pull request merged**